### PR TITLE
Swayfx

### PR DESCRIPTION
**Pull request by eclairevoyant**
Fixes swayfx breakages when bumping sway version

### CODE REVIEW

File: pkgs/by-name/sw/swayfx-unwrapped/package.nix [hunk 1]
```diff
@@ -18,6 +19,8 @@
     systemdSupport
     trayEnabled
     ;
+
+  wlroots_0_17 = wlroots_0_16;
```
Comment (raboof) [inline pkgs/by-name/sw/swayfx-unwrapped/package.nix:23]: :cry: maybe we should make it `wlroots` there after all then

**Review by raboof — APPROVED** [review]
I'm OK to merge this, but perhaps we should use `wlroots` instead of `wlroots_0_17` in sway-unwrapped then...

File: pkgs/by-name/sw/swayfx-unwrapped/package.nix (hunk 1 above)
Comment (eclairevoyant) [reply pkgs/by-name/sw/swayfx-unwrapped/package.nix:23]: Is there a guideline that prefers that? I don't see anything in contributing.md or pkgs/readme.md on that.

And to me, it's further obscuring the dependencies...

Comment (raboof) [reply pkgs/by-name/sw/swayfx-unwrapped/package.nix:23]: Neither is great, but passing `wlroots_0_16` for `wlroots_0_17` seems rather confusing to me...

Comment (thiagokokada) [reply pkgs/by-name/sw/swayfx-unwrapped/package.nix:23]: I think this ends up being the limitation of the current `pkgs/by-name` approach.

But really, I think we should just decouple `swayfx` from `sway` derivation instead. Copy-and-paste sometimes is better than having coupling between packages, and the fact that any change in `sway` can break `swayfx` is a strong argument to not having those 2 derivations share code.

File: pkgs/by-name/sw/swayfx-unwrapped/package.nix [hunk 2]
```diff
@@ -29,18 +32,6 @@
...
-  # remove when next release is rebased on Sway 1.9
-  patches =
-    let
-      removePatches = [
-        "LIBINPUT_CONFIG_ACCEL_PROFILE_CUSTOM.patch"
-      ];
-    in
-    builtins.filter
-      (patch: !builtins.elem (patch.name or null) removePatches)
-      (oldAttrs.patches or [ ]);
```
Comment (thiagokokada) [inline pkgs/by-name/sw/swayfx-unwrapped/package.nix:42]: I think this is another good example on why trying to reuse code sometimes ends up unnecessary complicating things.

I am in favor of just making those 2 derivations completely independent. WDYT @raboof @eclairevoyant ?

Comment (raboof) [reply pkgs/by-name/sw/swayfx-unwrapped/package.nix:42]: It's a balance between avoiding duplication and being autonomous.

For my own Sway fork, Volare, the build 'itself' is independent (https://codeberg.org/raboof/volare/src/branch/main/flake.nix#L10) but I do re-use the sway module (https://codeberg.org/raboof/volare#when-using-flakes-for-your-nixos-configuration).

Comment (thiagokokada) [reply pkgs/by-name/sw/swayfx-unwrapped/package.nix:42]: I think this approach is just causing an additional maintenance burden. Decoupling would solve all problems and avoid new ones in the future.

Also, if I remember correctly overriding packages causes a few strange issues, for example the update scripts don't work as it should (they try to update the original package instead the overriden one).

Comment (eclairevoyant) [reply pkgs/by-name/sw/swayfx-unwrapped/package.nix:42]: > Also, if I remember correctly overriding packages causes a few strange issues, for example the update scripts don't work as it should

I would call this a bug in the update script. But yes, let's eventually decouple these. They are fundamentally just different codebases, have different release schedules, etc. and have diverged substantially by now.

**Pull request merged**